src/commands/ai/registry.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright(c) ZenML GmbH 2024. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0(the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied.See the License for the specific language governing
+// permissions and limitations under the License.
+
+// TODO the registration of secrets commands should go in here, which is then imported into the ZenExtension file
+// In registry, the "register command thing is passed context (i.e. context.secrets) as an arg"
+
+import { aiCommands } from './cmds';
+import { registerCommand } from '../../common/vscodeapi';
+import { ZenExtension } from '../../services/ZenExtension';
+import { ExtensionContext, commands } from 'vscode';
+
+/**
+ * Registers GenAI related commands for the extension.
+ *
+ * @param {ExtensionContext} context - The context in which the extension operates, used for registering commands and managing their lifecycle.
+ */
+export const registerAICommands = (context: ExtensionContext) => {
+  try {
+    const registeredCommands = [
+      registerCommand('zenml.displayNextCodeRecommendation', () =>
+        aiCommands.displayNextCodeRecommendation()
+      ),
+      registerCommand('zenml.acceptCodeRecommendation', () =>
+        aiCommands.acceptCodeRecommendation()
+      ),
+    ];
+
+    registeredCommands.forEach(cmd => {
+      context.subscriptions.push(cmd);
+      ZenExtension.commandDisposables.push(cmd);
+    });
+  } catch (error) {
+    console.error('Error registering AI commands:', error);
+    commands.executeCommand('setContext', 'AICommandsRegistered', false);
+  }
+};
```

Comment:
For 1, I'm not sure if the redundancy can be consolidated or not... this was some boilerplate taken from the other command registration files that push the command to two locations, so it could be necessary for how this extension was built. I think we should leave it as-is to maintain consistency with the other commands.

Similarly for 2, I don't know if this is information that the end user needs; I believe if there is an issue with registering of these commands the current `console.error` should be sufficient. This is also how the other `registry.js` files handle errors.